### ISSUE TITLE
User-Agent: should allow computation of client success rates by platform and plugin.

### ISSUE THREAD

**Issue by pde** [letsencrypt/letsencrypt#1397]
In order to make strategic decisions about plugin prioritisation (engineering prioritisation, whether to ship or not ship each plugin in `letsencrypt-auto`, and which ones to prefer in the UI), we need to know which ones are succeeding for people.

That means we should send a small amount of plugin and platform information by default in the client's User Agent string, so that we can compute aggregate success rates on the server side.  Probably it should be which plugins are in use (`Authenticator: Standalone; Installer: None`) and platform major version (`Debian 8`). We should have a flag to disable it, for people who prefer not to reveal anything at all about their server software.

**Comment by pde** [comment]
Flagging this urgent because the sooner it's shipped, the sooner we can prioritise 1.0 tickets for Apache vs webroot (vs standalone?).

**Comment by pde** [comment]
In terms of implementation:
* we  learn the plugins [here](https://github.com/letsencrypt/letsencrypt/blob/master/letsencrypt/cli.py#L426). You get their names by doing something like `authenticator.name`, provided authenticator is not `None`.
* we do something clever to determine the platform and major version
* we can pass them over to the `acme` module as optional keyword arguments, so as not to break uses of that module that don't pass along User Agent information.
* we ensure that acme's [_send_requests](https://github.com/letsencrypt/letsencrypt/blob/master/acme/acme/client.py#L561) helper uses the User Agent we want.

**Comment by pde** [comment]
@yaseminacar @sfahl

**Comment by kuba** [comment]
Duplicate of #858.

**Comment by xamnesiax-sx** [comment]
@pde https://github.com/xamnesiax-sx/letsencrypt/commit/9de2cc0f2480f1e22458377d126c32a22bafafd8 I've mostly implemented it, need some clarification on a few things. I'll hit you up on IRC tomorrow

**Comment by pde** [comment]
Here is what the final User Agent should perhaps look like:

```
User-Agent: LetsEncryptPythonClient/0.1.0.dev0 (Debian 8) Authenticator/standalone Installer/none
```

**Comment by pde** [comment]
The client version ("0.1.0dev0" in my example above) can be found [here](https://github.com/letsencrypt/letsencrypt/blob/master/letsencrypt/__init__.py).

**Comment by pde** [comment]
Closed in #1455 .  Thanks for your help on this @xamnesiax-sx and @bmw !
